doc/sequence-diagram.puml

```diff
@@ -58,23 +60,32 @@ group 4. HotShot consensus
     ValidatedState -> ValidatedState: apply header\ncharge fee
 end group
 
-group 5. HotShot proving
+group 5. Rollup block production
+    query -> executor: Espresso header,\ntransactions in namespace
+    executor -> executor: verify\nnamespace\nproof
+    executor -> rprover: rollup block
+end group
+
+group 6. HotShot proving
     signer -> relay: state, signature
-    relay -> prover: state, signature bundle
+    relay -> prover: state,\nsignature bundle
     prover -> "Light\nClient\nContract" as LC: proof
     LC -> LC: verify
 end group
 
-group 6. Rollup proving
-    loop N (rollup blocks per proof) times
-    query -> executor: Espresso header,\ntransactions in namespace
-    executor -> executor: verify namespace proof
-    executor -> rprover: rollup block
-                           end
+group 7. Rollup proving
     rprover -> RC: rollup state proof
 
     group verify
         LC -> "Rollup\nContract" as RC: HotShot\ncommitment
     end group
 end group
+
+group 8. Rollup dispute
```

Comment:
In the case the challenger is right, what happens at the end of the dispute period? Is the last rollup state reverted?

Reply:
Not sure actually.